### ISSUE TITLE
Export Layer Group as Animated GIF

### ISSUE THREAD

**Issue by Andrake-A** [kamilburda/gimp-export-layers#110]
Would it be possible to add an option to export each layer in a layer group as an animated gif? 

I know I can already export a layer group as one still image but when the 'Export Image as GIF' dialog pops up the 'As animation' box is greyed out for some reason as well as all the other animated gif options.

example, I have a layer structure like this
```
├── image.jpg
├── image.jpg
├── animated image.gif
│   ├── frame 1
│   ├── frame 2
│   ├── frame 3
```
and that would three images with the third one being animated.

Is this already achievable or am I requesting a new feature?

**Comment by kamilburda** [comment]
Try if the following works for you:

1. Add the "Export" procedure. The folder explorer and the file extension in the main dialog will now be ignored.
2. Adjust the output folder as needed.
3. Type `gif` next to `File extension`.
4. Set `Perform export:` to `For each top-level layer or group`.

Once you start exporting and your first item is a layer group, you should be able to check `As animation`. However, if the first item is a layer, it seems the option cannot be checked and you will not be able to adjust this for subsequent layers. You could rearrange your image or add a dummy layer group to work around this.

If each layer within a layer group has different dimensions, this could be a problem. As a workaround, you could uncheck "Use layer size" to use the image size, but not this will use the same size for all layers within the image.

**Comment by Andrake-A** [comment]
Thanks! Your solution worked. However, the two non-animated layers I showed in my example are made into .gif files also insted of .jpg when using the "Export" procedure. I checked "Use file extension in layer name" but that didn't seem to do anything. It might be a bug, I dont know.

**Comment by kamilburda** [comment]
Type `[layer name, %e]` in the main dialog next to `Save as:`. This should preserve the file extension.

**Comment by Andrake-A** [comment]
Okay, that worked. Thanks!

**Comment by kamilburda** [comment]
This is definitely a bug, and I can confirm it occurs on my end as well.

In the meantime, you could work around this by performing export twice - first with the layer group(s) only and then with the standalone layers.

To export the layer group only, uncheck the `Layers` constraint and add the `Layer groups` constraint.

To export the regular layers only, uncheck the `Layer groups` constraint and check the `Layers` constraint.

**Comment by Andrake-A** [comment]
I'll try. Still, I appreciate your support anyways.

**Comment by kamilburda** [comment]
I just released version 4.1, which should fix the export.

**Comment by Andrake-A** [comment]
OK, I tried it, and everything seems to come out how I want it. Thanks for the help!
